src/combinators/mod.rs

```diff
@@ -932,6 +973,32 @@ pub fn tuple<W: Write, List: Tuple<W>>(l: List) -> impl SerializeFn<W> {
     }
 }
 
+/// Reserves space for the `Before` combinator, applies the `Gen` combinator,
+/// then applies the `Before` combinator with the output from `Gen` onto the
+/// reserved space.
+///
+/// ```rust
+/// use cookie_factory::{tuple, back_to_the_buffer, string, be_u8, be_u32, WriteCounter};
+///
+/// let mut buf = [0; 9];
+/// tuple((
+///     back_to_the_buffer(
+///         4,
+///         &move |buf| string("test")(WriteCounter::new(buf)).map(|counter| counter.into_inner()),
+///         &move |buf, len| be_u32(len as u32)(buf)
+///     ),
+///     be_u8(42)
+/// ))(&mut buf[..]).unwrap();
+/// assert_eq!(&buf, &[0, 0, 0, 4, 't' as u8, 'e' as u8, 's' as u8, 't' as u8, 42]);
+/// ```
+pub fn back_to_the_buffer<'a, W: BackToTheBuffer, Tmp, Gen: 'a, Before: 'a>(reserved: usize, gen: &'a Gen, before: &'a Before) -> impl SerializeFn<W> + 'a
```

Comment:
Why do we take references to the `Gen` / `Before` closures here instead of taking them by value?

Reply:
I wish we had another way around this (maybe you know of one?), but since we are using `Fn` and not `FnOnce`, we cannot move values in the closure, they need to be either cloned, or to implement `Copy`, `Fn` do neither

Reply:
Oh, it seems to work, taking values here, as long as the trait method takes references

Reply:
:tada: Nicer :)